calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino

```diff
@@ -79,69 +57,75 @@ void setup() {
-    extensionCount_[i]=counter_[i];
-    Serial.print(extensionCount_[i]);
-    Serial.print("\t\t\t\t");  
+    extensionCount[i]=counter[i];
+    Serial.print("\t\t");  
+    Serial.print(extensionCount[i]);
+    Serial.print("\t\t\t");  
   }
   delay(1000);
   
   moveTillLimit(-1, 255); //retract fully and count pulses
-  Serial.print("\nRetraction Count:\t\t");
+  Serial.print("\nRetraction Count:");
   for(int i=0; i<numberOfActuators; i++){
-    retractionCount_[i]=counter_[i];
-    Serial.print(retractionCount_[i]);
-    Serial.print("\t\t\t\t");  
+    retractionCount[i]=counter[i];
+    Serial.print("\t\t");  
+    Serial.print(retractionCount[i]);
+    Serial.print("\t\t\t");  
   }
   delay(1000);
 
   for(int i=0; i<numberOfActuators; i++){
     Serial.print("\nActuator ");
     Serial.print(i);
     Serial.print(" average pulses: ");
-    pulseTotal[i]=(extensionCount_[i]+retractionCount_[i])/2;   //takes the average of measurements
+    pulseTotal[i]=(extensionCount[i]+retractionCount[i])/2;   //takes the average of measurements
     Serial.print(pulseTotal[i]);  
-  }
-  
-  Serial.println("\nEnter these values in the syncronous control progam.");
+  }  
+  Serial.println("\n\nEnter these values in the synchronous control progam.");
+}
+
+void loop() {  
 }
 
 void moveTillLimit(int Direction, int Speed){
-  //reset counter variables
+  //this function moves the actuator to one of its limits
   for(int i = 0; i < numberOfActuators; i++){
-    counter[i] = 0;
+    counter[i] = 0;                                     //reset counter variables
     prevCounter[i] = 0;
   }  
   do {
-    //keep move until counter remains the same for 100ms (i.e at the end)  
     for(int i = 0; i < numberOfActuators; i++) {
-      prevCounter_[i] = counter_[i];
+      prevCounter[i] = counter[i];
     }
     timeElapsed = 0;
-    while(timeElapsed < period){
+    while(timeElapsed < period){                       //keep moving until counter remains the same for the duration of "period" variable  
       for(int i = 0; i < numberOfActuators; i++) {
         driveActuator(i, Direction, Speed);
       }
     }
-  } while(compareCounter(prevCounter, counter))
+  } while(compareCounter(prevCounter, counter));      //loop until all counts remain the same
 }
 
-// If both arrays have a similar element in the same index than return false
-bool compareCounter(int prevCounter, int counter){
-  bool areEqual = true;
+bool compareCounter(int prevCounter[], int counter[]){
+  //compares two arrays and returns false when every element of one array is the same as its corresponding indexed element in the other array
+  bool areUnequal = true;
   for(int i = 0; i < numberOfActuators; i++){
-    if(prevCounter[i] == counter[i] ){
-      areEqual = false;
+    if(prevCounter[i] == counter[i]){
+      areUnequal = false;
+    } 
+    else{                                             //if even one pair of elements are unequal the entire function returns true
```

Comment:
additional condition to ensure all elements satisfy condition, not just one pair